runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java

```diff
@@ -206,45 +242,59 @@ public void onChatMessage(ChatMessage chatMessage)
 			}
 		}
 
-		if (highlightMatcher != null)
+		if (!highlightPatterns.isEmpty())
 		{
+
+			boolean foundAtLeastOne = false;
+			// Get nodeValue to store and update in between the different pattern passes
+			// The messageNode value is only set after all patterns have been processed
 			String nodeValue = messageNode.getValue();
-			Matcher matcher = highlightMatcher.matcher(nodeValue);
-			boolean found = false;
-			StringBuffer stringBuffer = new StringBuffer();
 
-			while (matcher.find())
+			for (Pattern pattern : highlightPatterns)
 			{
-				String value = matcher.group();
+				boolean found = false;
+				// Reset stringBuffer at each pattern to update message highlights iteratively
```

Comment:
It's being reinitialized, not reset. This comment isn't correct or particularly useful, let's remove it.
```suggestion
```